lib/compress/zstd_lazy.c

```diff
@@ -921,6 +921,28 @@ static ZSTD_VecMask ZSTD_Vec256_cmpMask8(ZSTD_Vec256 x, ZSTD_Vec256 y) {
   sndMask = ZSTD_Vec128_cmpMask8(x.snd, y.snd);
   return fstMask | (sndMask << 16);
 }
+#else//AVX2 
+typedef struct {
+	__m256i v;
+} ZSTD_Vec256;
+
+static ZSTD_Vec256 ZSTD_Vec256_read(const void* const ptr) {
+	ZSTD_Vec256 v;
+    v.v = _mm256_loadu_si256((const __m256i*)ptr);
+	return v;
+}
+
+static ZSTD_Vec256 ZSTD_Vec256_set8(BYTE val) {
+	ZSTD_Vec256 v;
+    v.v = _mm256_set1_epi8((char)val);
+	return v;
+}
+
+static ZSTD_VecMask ZSTD_Vec256_cmpMask8(ZSTD_Vec256 x, ZSTD_Vec256 y) {
+    return (ZSTD_VecMask)_mm256_movemask_epi8(_mm256_cmpeq_epi8(x.v, y.v));
+}
+
+#endif
```

Comment:
Can you submit this portion as a separate PR please?

It should be good go go with some minor formatting changes, but this portion requires performance testing, where the MSVC fix is easy to accept.

I know I measured the performance of AVX2 vs SSE, and it was slightly better. But, I didn't see a huge speed-up.

Reply:
Yea I'll remove this and submit it seperate